### PR TITLE
New take on cross-building

### PR DESCRIPTION
**Pull request by regadas**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,25 +1,18 @@
...
-    - scala: 2.11.12
-      jdk: oraclejdk8
-    - scala: 2.12.6
-      jdk: oraclejdk8
+
+jdk: oraclejdk8
+
+env:
+  - SBT_CMD="scalafmtSbtCheck scalafmtCheck test:scalafmtCheck scalastyle test:scalastyle"
+  - SBT_CMD="coverage +test coverageReport coverageAggregate"
```
Comment (nevillelyh) [inline .travis.yml]: The 2nd line overwrites the first?

File: build.sbt [hunk 2]
```diff
@@ -156,32 +158,24 @@ lazy val core: Project = project
...
-    projectDependencies := {
-      if (scalaBinaryVersion.value == "2.12") Nil
-      else
-        Seq(
-          (projectID in core).value,
-          (projectID in tensorflow).value,
-          (projectID in xgboost).value
-        )
-    }
+    projectDependencies := Seq(
```
Comment (nevillelyh) [inline build.sbt]: We can probably get rid of these and rely on `.dependsOn` only?

File: .travis.yml (hunk 1 above)
Comment (regadas) [reply .travis.yml]: this will build a env matrix; those `SBT_CMD` will run in parallel.

File: build.sbt (hunk 2 above)
Comment (regadas) [reply build.sbt]: done.

File: .travis.yml [hunk 3]
```diff
@@ -1,25 +1,21 @@
...
+
+jdk: oraclejdk8
+
 matrix:
-  include:
-    - scala: 2.11.12
-      jdk: oraclejdk8
-    - scala: 2.12.6
-      jdk: oraclejdk8
+  fast_finish: true
```
Comment (regadas) [inline .travis.yml]: this will make sure that if one of the steps fail the build will fail fast.

**Pull request merged**